### PR TITLE
chore: support provision tab app resources using arm

### PR DESCRIPTION
**Pull request by blackchoey**
With this PR, we can provision tab app resources using ARM when related feature flag is turned on

### CODE REVIEW

File: packages/fx-core/src/common/tools.ts [hunk 1]
```diff
@@ -533,6 +533,6 @@ export function compileHandlebarsTemplateString(templateString: string, context:
 
 export function getArmOutput(ctx: PluginContext, key: string): string | undefined {
   const solutionConfig = ctx.configOfOtherPlugins.get("solution");
-  const output = solutionConfig?.get("armTemplate");
+  const output = solutionConfig?.get("armTemplateOutput");
```
Comment (dilin-MS) [inline packages/fx-core/src/common/tools.ts:536]: Please use the constant value `ARM_TEMPLATE_OUTPUT` in *packages\fx-core\src\plugins\solution\fx-solution\constants.ts*.

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 2]
```diff
@@ -122,19 +122,30 @@ export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<v
...
-  const parameterJson = JSON.parse(expandParameterPlaceholders(ctx, parameterTemplate));
   const parameterDefaultFilePath = path.join(
     azureInfraDir,
     parameterFolder,
     parameterDefaultFileName
   );
-  await fs.writeFile(parameterDefaultFilePath, parameterJson);
+  let parameterTemplate, parameterJsonString;
+  try {
+    await fs.stat(parameterDefaultFilePath);
```
Comment (dilin-MS) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:135]: What about check whether the `parameterDefaultFilePath` exists first. If yes, read the file. Else, read the parameterTemplateFile.

Comment (dilin-MS) [reply packages/fx-core/src/plugins/solution/fx-solution/arm.ts:135]: What if fail to read parameter from these two files?

File: packages/fx-core/src/plugins/resource/frontend/configs.ts [hunk 3]
```diff
@@ -72,6 +72,9 @@ export class FrontendConfig {
     let storageName: string | undefined;
     if (isArmSupportEnabled()) {
       storageName = getArmOutput(ctx, ArmOutput.FrontendStorageName) as string;
+      if (!storageName) {
```
Comment (hund030) [inline packages/fx-core/src/plugins/resource/frontend/configs.ts:75]: The logic is quiet complex. Ok for now but we better to improve it.

**Review by hund030 — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/frontend/configs.ts (hunk 3 above)
Comment (blackchoey) [reply packages/fx-core/src/plugins/resource/frontend/configs.ts:75]: yes, I think we need to do some refractor to split the provision and deploy related logic here. Let's do this later.

**Review by 1yefuwang1 — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/frontend/configs.ts (hunk 3 above)
Comment (dilin-MS) [reply packages/fx-core/src/plugins/resource/frontend/configs.ts:75]: Suggest add "todo" comment to avoid being forgotten.

**Review by KennethBWSong — APPROVED** [review]

**Pull request merged**